### DISCUSSION TITLE
Scoring a trace after the LLM chain returns

### DISCUSSION THREAD

**Discussion by pooja1423** [langfuse/langfuse#1610] [Support] [answered]
Hello, I am trying to score a trace based on the output of a Langchain LLM chain. I am unable to associate a score with the trace and it doesn't show up in the web UI. However, if I call score on the trace before the LLM chain is invoked, the score shows up in the UI. How do I get around this? I am using the python SDK like so:

`trace = langfuse.trace(session_id=session_id, metadata=metadata, tags=tags)
  handler = trace.get_langchain_handler()

 result = llm_chain.invoke(config={"callbacks": [handler])
 langfuse.score(trace_id=trace.id, name="pre-score", value=1 if result else 0)`

    `

**Comment by noble-varghese** [comment, marked as answer]
Hi @pooja1423 
I had a similar issue in a slightly different context. But in general langfuse executes network requests in the background for better performace without blocking the current execution thread. To ensure that the requests are completely sent before terminating the application it is suggested to add `langfuse.flush()` at the end of the script or on shutdown. You can read more [here](https://langfuse.com/docs/sdk/python/low-level-sdk#additional-configurations)
Hope this helps :)

**Reply by pooja1423** [reply]
That did it, thank you!
